src/coalesce-vue-vuetify3/src/components/input/c-datetime-picker.vue

```diff
@@ -584,6 +594,34 @@ function emitInput(value: Date | null) {
   }
 }
 
+function setToday() {
+  const now = new Date();
+  const today = startOfDay(now);
+
+  // Preserve existing time if it's set
+  if (internalValueZoned.value) {
+    const existingDate = internalValueZoned.value;
+
+    const newDate = set(today, {
+      hours: existingDate.getHours(),
+      minutes: existingDate.getMinutes(),
+      seconds: existingDate.getSeconds(),
+      milliseconds: existingDate.getMilliseconds(),
+    });
+
+    if (internalTimeZone.value) {
+      // Adjust for the specified time zone
+      modelValue.value = fromZonedTime(newDate, internalTimeZone.value);
+    } else {
+      // No time zone specified, use the new date as is
+      modelValue.value = newDate;
+    }
+  } else {
+    // No existing date/time, just set to start of today
+    modelValue.value = today;
+  }
```

Comment:
This implementation:
- Bypasses min/max/allowedDates/step enforcement
- Doesn't bind to `valueOwner` (i.e. through `:model :for` binding)
- Duplicates time preservation logic

Fortunately, its an easy fix:

```suggestion
  dateChanged(new Date())
```